blobfuse/blobfuse.cpp

```diff
@@ -1331,3 +1378,99 @@ int initialize_blobfuse()
     globalTimes.lastModifiedTime = globalTimes.lastAccessTime = globalTimes.lastChangeTime = time(NULL);
     return 0;
 }
+
+
+// Code to generate json file for rustfuse and launch the binary
+int mount_rust_fuse(char* argv[]){
+    json j;
+    if (config_options.authType == SPN_AUTH){
+        // adlsgen1fuse will be reading secret from env variable (ADL_CLIENT_SECRET) hence no reason to include this.
+        //j["clientsecret"] = config_options.spnClientSecret;
+
+        j["clientid"] = config_options.spnClientId;
+        j["tenantid"] = config_options.spnTenantId;
+        if(config_options.aadEndpoint != "")
+        {
+            j["authorityurl"] = config_options.aadEndpoint;
+        }
+        else
+        {
+            j["authorityurl"] = "https://login.microsoftonline.com";
+        }
+
+        j["credentialtype"] = "servicePrincipal";
+    } else{
+        syslog(LOG_ERR, "For Gen1 account only SPN auth is supported.");
+        return -1;
+    }
+
+    j["resourceurl"] = "https://datalake.azure.net/";
```

Comment:
here , pls add a check to see if the container name is not empty. If it is not empty then append containername + "/" at the end. ADLS gen1 folder works similar to a container so lets just keep that variable name.

Reply:
append to resourceurl ? or resourceid?

Reply:
Does customer really need a feature to mount a specific folder within container. If such feature is not there in existing solution lets skip this part. Also if we decide to add this anyway then shouldn't resourceID also needs similar modification ?

Reply:
It is there, meaning they can mount root  - empty container, also means the whole account
or
they can mount subfolders , we can use the container-name for subfolder.
I have commented on the resourceid also to append the container name if any